File: CleverFerretV2/feature/metadata/src/main/java/com/cleverferret/v2/feature/metadata/api/MetadataFeatureApi.java
package com.cleverferret.v2.feature.metadata.api;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.metadata.services.metadata.MetadataLookupResultV1;

public interface MetadataFeatureApi {
    String featureKey();

    IntegrationResult<MetadataLookupResultV1> fetchMetadata(String query);
}

File: CleverFerretV2/feature/metadata/src/main/java/com/cleverferret/v2/feature/metadata/api/MetadataFeatureApi.kt
package com.cleverferret.v2.feature.metadata.api

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.metadata.services.metadata.MetadataLookupResultV1

interface MetadataFeatureApi {
    fun featureKey(): String
    fun fetchMetadata(query: String): IntegrationResult<MetadataLookupResultV1>
}
